app/models/course/assessment.rb

```diff
@@ -135,10 +135,18 @@ def initialize_duplicate(duplicator, other)
     copy_attributes(other, duplicator.time_shift)
     self.folder = duplicator.duplicate(other.folder)
     self.questions = duplicator.duplicate(other.questions.map(&:actable)).compact.map(&:acting_as)
-    self.assessment_conditions = duplicator.duplicate(other.assessment_conditions)
-    # Like achievement conditions, duplicate the actable object directly and let the acting_as
-    # gem create the Condition object.
-    self.conditions = duplicator.duplicate(other.conditions.map(&:actable)).map(&:acting_as)
+    if duplicator.mode == :course
+      self.assessment_conditions = duplicator.duplicate(other.assessment_conditions)
+      # Like achievement conditions, duplicate the actable object directly and let the acting_as
+      # gem create the Condition object.
+      self.conditions = duplicator.duplicate(other.conditions.map(&:actable)).map(&:acting_as)
+    elsif duplicator.mode == :object
+      target_category = duplicator.options[:target_course].assessment_categories.first
```

Comment:
I meant something like:
```
def initialize_duplicate(*other_args, target_course = nil)
   if target_course
     xxx
end
```

Because now `initialize_duplicate` know the mode/options of the duplicator, I was trying to decouple them and not let the model directly access the options of the duplicator.